IO.go

```diff
@@ -64,3 +77,43 @@ func (io IO) SetDeltaTime(value float32) {
 func (io IO) SetFontGlobalScale(value float32) {
 	C.iggIoSetFontGlobalScale(io.handle, C.float(value))
 }
+
+// KeyPress sets the KeysDown flag
+func (io IO) KeyPress(key int) {
+	C.iggIoKeyPress(io.handle, C.int(key))
+}
+
+// KeyRelease clear the KeysDown flag
+func (io IO) KeyRelease(key int) {
+	C.iggIoKeyRelease(io.handle, C.int(key))
+}
+
+// KeyMap maps a key into the KeysDown array which represents your "native" keyboard state
+func (io IO) KeyMap(imguiKey int, glfwKey int) {
+	C.iggIoKeyMap(io.handle, C.int(imguiKey), C.int(glfwKey))
+}
+
+// KeyCtrl sets the keyboard modifier control pressed
+func (io IO) KeyCtrl(leftCtrl int, rigthCtrl int) {
+	C.iggIoKeyCtrl(io.handle, C.int(leftCtrl), C.int(rigthCtrl))
+}
+
+// KeyShift sets the keyboard modifier shift pressed
+func (io IO) KeyShift(leftShift int, rigthShift int) {
+	C.iggIoKeyShift(io.handle, C.int(leftShift), C.int(rigthShift))
+}
+
+// KeyAlt sets the keyboard modifier alt pressed
+func (io IO) KeyAlt(leftAlt int, rigthAlt int) {
+	C.iggIoKeyAlt(io.handle, C.int(leftAlt), C.int(rigthAlt))
+}
+
+// KeySuper sets the keyboard modifier super pressed
+func (io IO) KeySuper(leftSuper int, rigthSuper int) {
+	C.iggIoKeySuper(io.handle, C.int(leftSuper), C.int(rigthSuper))
+}
+
+// AddInputCharacter adds a new character into InputCharacters[].
+func (io IO) AddInputCharacter(char rune) {
+	C.iggIoAddInputCharacter(io.handle, C.ushort(char))
```

Comment:
Instead of the AddInputCharacter, I would use the AddInputCharactersUTF8 function of ImGui::IO .
Reason: this one takes an UTF-8 array, which incidentally also Go uses for strings.

So, please change it to the Go function AddInputCharacters(chars string) and route it through to ImGui to AddInputCharactersUTF8.
This then removes the necessity to "filter" runes outside uint16 limit. and the GLFW handler could then pass in the to-string-converted rune.